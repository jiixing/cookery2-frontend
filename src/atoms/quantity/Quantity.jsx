// @flow

import React from 'react'

import type { Quantity as QuantityType } from 'types'

const Quantity = ({ name, unit, amount }: QuantityType) => (
  <div>
    <span className="text-capitalize">{name}</span>
    <span className="font-italic float-right">{`${amount} ${unit}`}</span>
  </div>
)

export default Quantity
