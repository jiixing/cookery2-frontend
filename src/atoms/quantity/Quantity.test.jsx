import React from 'react'
import renderer from 'react-test-renderer'

import Quantity from './Quantity'

describe('components', () => {
  describe('Quantity', () => {
    it('renders correctly', () => {
      const component = renderer.create(
        <Quantity name="Olive oil" unit="ml" amount={50} />
      )
      expect(component.toJSON()).toMatchSnapshot()
    })
  })
})
