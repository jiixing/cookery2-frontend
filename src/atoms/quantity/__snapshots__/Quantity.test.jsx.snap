// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components Quantity renders correctly 1`] = `
<div>
  <span
    className="text-capitalize"
  >
    Olive oil
  </span>
  <span
    className="font-italic float-right"
  >
    50 ml
  </span>
</div>
`;
